src/processing.ts

```diff
@@ -177,3 +177,93 @@ export const min = (a: bigint, b: bigint) => (a < b ? a : b);
  * const biggest = max(5n, 10n) // 10n
  */
 export const max = (a: bigint, b: bigint) => (a > b ? a : b);
+
+/**
+ * Register the proposals as as active (to enable later identification
+ * of answers) and issue notifications for them.
+ *
+ * @param space - The space that the proposals belong to
+ * @param proposals - The proposals to process
+ *
+ * @example
+ * await processProposals(space, proposals)
+ */
+export const processProposals = (space: Space, proposals: ProposalQuestionCreated[]) => {
+  return configurableProcessProposals({
+    space,
+    proposals,
+    notifyFn: notify,
+    insertProposalFn: insertProposal,
+  });
+};
+
+type ConfigurableProcessProposalsDeps = {
+  space: Space;
+  proposals: ProposalQuestionCreated[];
+  notifyFn: typeof notify;
+  insertProposalFn: typeof insertProposal;
+};
+export const configurableProcessProposals = async (deps: ConfigurableProcessProposalsDeps) => {
+  const { space, proposals, insertProposalFn, notifyFn } = deps;
+
+  await Promise.all(
+    proposals.map((event) =>
+      Promise.all([
+        insertProposalFn({
+          ens: space.ens,
+          proposalId: event.proposalId,
+          txHash: event.txHash,
+          questionId: event.questionId,
+          happenedAt: event.happenedAt,
+        }),
+        notifyFn({
+          type: EventType.PROPOSAL_QUESTION_CREATED,
```

Comment:
So we make a tuple-[insertProposalFn,notifyFn] for each event. Wrap it in Promise.all, so it's like:
Promise.all([insertProposalFn,notifyFn])

This map into a larger promises array

[[tuple],[tuple],[tuple]]

And then we await this

Is that the order?

Reply:
I think you got it right, but the syntax gives some room to mistakes (`[[tuple],[tuple],[tuple]]` looks like an array of arrays instead of `[promise (waiting a tuple), promise (waiting a tuple), promise (waiting a tuple)]).

My intuition is that I create a Promise first for handling a proposal. That consists, right now, in `insertProposal` and `notify`. We can call that promise  (for example) `proposalPromise`. Technically, as you said, it is a tuple passed to `Promise.all`.

As that process is in a map, I have an array of `proposalPromise`. An array of promises that handles the individual proposals. The code finally wait for the resolution of that array of `proposalPromise` to know when everything has been processed.